### PR TITLE
Test integration with `BlockArray`

### PR DESCRIPTION
**Pull request by jofrevalles**
### Summary
This PR adds integration tests for the `BlockArray` library, addressing the issue #8. The tests cover operations such as `contract`, `permutedims`, and more. Here is a summary of the changes:
- Implement tests for creating tensors with block arrays
- Implement tests for `permutedims` operation with block arrays
- Implement tests for `contract` operation with block arrays

Unfortunately, I couldn't include a test for the `SVD` decomposition, since `BlockArray` does not support it. The relevant issue discussing the lack of `SVD` support (and apparently more linear algebra functions) in `BlockArrays.jl` can be found here: https://github.com/JuliaArrays/BlockArrays.jl/issues/131

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -20,4 +20,4 @@ julia = "1.8"
 ChainRulesCoreExt = "ChainRulesCore"
 
 [weakdeps]
-ChainRulesCore = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
\ No newline at end of file
+ChainRulesCore = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
```
Comment (mofeing) [inline Project.toml:23]: What is the modification here?

File: test/integration/BlockArray_test.jl [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        tensor = Tensor(data1, [:i, :j])
+        block_tensor = Tensor(block_array, [:j, :k])
+
+        contracted_tensor = contract(tensor, block_tensor)
+
+        @test contracted_tensor |> labels == (:i, :k)
+        @test (contracted_tensor |> parent |> blocksizes)[2] == [2, 2]
+        @test Array(parent(contracted_tensor)) ≈ parent(contract(tensor, Tensor(data2, [:j, :k])))
+    end
```
Comment (mofeing) [inline test/integration/BlockArray_test.jl:71]: Group test sets into a "contract" test set.
```suggestion
    @testset "contract block tensors" begin
        data1, data2 = rand(4, 4), rand(4, 4)
        block_sizes1, block_sizes2 = ([3, 1], [2, 2]), ([1, 3], [2, 2])
        block_array1 = BlockArray(data1, block_sizes1...)
        block_array2 = BlockArray(data2, block_sizes2...)

        tensor1 = Tensor(data1, [:i, :j])
        tensor2 = Tensor(data2, [:j, :k])
        block_tensor1 = Tensor(block_array1, [:i, :j])
        block_tensor2 = Tensor(block_array2, [:j, :k])

        contracted_tensor = contract(tensor1, tensor2)
        contracted_block_tensor = contract(block_tensor1, block_tensor2)

        @test parent(contracted_block_tensor) isa BlockArray
        @test contracted_block_tensor |> labels == (:i, :k)
        @test contracted_block_tensor |> blocksizes == ([3, 1], [2, 2])
        @test Array(parent(contracted_block_tensor)) ≈ parent(contracted_tensor)
    end

    @testset "contract block-unblocked tensors" begin
        data1, data2 = rand(4, 4), rand(4, 4)
        block_sizes = ([3, 1], [2, 2])
        block_array = BlockArray(data2, block_sizes...)

        tensor = Tensor(data1, [:i, :j])
        block_tensor = Tensor(block_array, [:j, :k])

        contracted_tensor = contract(tensor, block_tensor)

        @test contracted_tensor |> labels == (:i, :k)
        @test (contracted_tensor |> parent |> blocksizes)[2] == [2, 2]
        @test Array(parent(contracted_tensor)) ≈ parent(contract(tensor, Tensor(data2, [:j, :k])))
    end
```

**Review by mofeing — CHANGES_REQUESTED** [review]

File: Project.toml (hunk 1 above)
Comment (jofrevalles) [reply Project.toml:23]: I don't understand why git counted that as a change, do you see any change?

Comment (jofrevalles) [reply Project.toml]: Done :+1:

**Pull request merged**